util/file.go

```diff
@@ -140,3 +150,19 @@ func mkUniq(prefix string, create func(string) error) (string, error) {
 
 	return "", err
 }
+
+// GetVersion returns the version of the binary file. It distinguishes between two cases:
+// a) if the current commit is the same commit of the last tag it returns the tag's name.
+// b) if the current commit differs from the commit of the last tag it returns
+// the short commit hash of the actual commit appended to the "dev" and date of commit.
+func GetVersion() (string, error) {
```

Comment:
Should not compute the version number during runtime, this logic should be in the Makefile during the build and linking process. (And also increase the PATCH version part for dev Versions.)